components/calendar/src/chinese.rs

```diff
@@ -379,6 +388,22 @@ impl ChineseBased for Chinese {
     }
 
     const EPOCH: RataDie = CHINESE_EPOCH;
+
+    fn get_compiled_data_for_year(year: i32) -> Option<ChineseBasedCompiledData> {
+        if (chinese_data::MIN_YEAR..=chinese_data::MAX_YEAR).contains(&year) {
+            #[allow(clippy::indexing_slicing)]
+            // indexing is ok because the if statement checks for bounds
+            let packed_data =
+                chinese_data::CHINESE_DATA_ARRAY[(year - chinese_data::MIN_YEAR) as usize];
```

Comment:
nit: this is better written as `if let Some(data) = ARRAY.get(year - MIN_YEAR)` to avoid the if + indexing